### DISCUSSION TITLE
x-mask breaking form in livewire

### DISCUSSION THREAD

**Discussion by DonCamillo11** [alpinejs/alpine#4023] [7. Bugs (archived)]
Hi,
I'm facing issues when using x-mask and wire:model in Livewire. Sending forms with _x-mask:dynamic="$money($input, ',')"_ and wire:model I get the error _Warning: 1265 Data truncated for column 'xxx' at row 1_. Same with _x-mask="99.99.9999"_ for dates. After submitting I get the error _Invalid datetime format: 1292 Incorrect date value: '' for column 'xxx' at row 1_. In both cases the fields are empty.

**Comment by shubdragon** [comment]
are you using some kind of date picker js library ? I had a same issue with dynamic input inside x-for loop but i solved it other way, i used x-model instead of wire:model and entangled whole array with livewire

**Comment by DonCamillo11** [comment]
No, just alpine.js. I solved it setting the variable of the date to null and the number as integer before saving. But I thought there might be a better solution. I don’t know how the fields are handled but It seems like the data is in the wrong format even though the fields are empty.

**Comment by ekwoka** [comment]
99.99.9999 is not a valid datetime format on a date input

I'm not sure about the first one. I don't understand the context.

**Comment by DonCamillo11** [comment]
Yes, I know but if the field is empty it should be null or am I wrong?

**Reply by ekwoka** [reply]
what should be null?

**Reply by DonCamillo11** [reply]
the value that I get from the date field. 
I have an input field "birthday". Without x-mask the empty field is submitted as a null value and everything is fine. When using x-mask the format of the "birthday" field is "" and it doesn't pass the validation.

**Reply by DonCamillo11** [reply]
same for the numeric values with _x-mask:dynamic="$money($input, ',')"_ . An empty value doesn’t pass through the numeric validation anymore

**Reply by ekwoka** [reply]
validation provided by what?

the values of those are normally not null when they are initialized with no values, iirc.

**Reply by DonCamillo11** [reply]
Sorry, I don’t understand your comment. Usually empty input fields are submitted as null values (checked by output after submitting the form). My birthday field in the database accepts a null value, so there is no problem. But with x-mask it is not null, so I get a QueryException (see above).

**Reply by ekwoka** [reply]
That's not true though, is what I mean.

an input with no value is sent in the FormData as an empty string, not `null`.

> value
> A string that represents the current value of the control. If the user enters a value different from the value expected, this may return an empty string.

https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement

I checked and yes, inputs have a value of `''`, not `null`

**Reply by DonCamillo11** [reply]
Sorry, thats not true, using livewire and Laravel an empty input field is sent as a null value.
That’s the ouput I get after sending an empty field. But with x-mask it seems to be an empty string.
![Bildschirmfoto 2024-02-09 um 14 15 00](https://github.com/alpinejs/alpine/assets/125701444/c7dd4c5a-b1ba-4f2e-91df-c29b2fbc4a30)

**Reply by ekwoka** [reply]
That's well and good.

What I described is how HTML behaves.

This seems lie an issue with Laravel/Livewire not respecting how HTML behaves in one way or another.

I'm not the decision maker, but I'd say Alpine should not be making special concessions to work with Laravel/Livewire that take it out of spec with HTML. It should be addressed on the Laravel/Livewire side.
